plugins/download_statistics/plugin_tests/download_statistics_test.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright Kitware Inc.
+#
+#  Licensed under the Apache License, Version 2.0 ( the "License" );
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+###############################################################################
+
+
+import os
+import json
+
+from tests import base
+from girder.constants import ROOT_DIR
+
+
+def setUpModule():
+    base.enabledPlugins.append('download_statistics')
+    base.startServer()
+
+
+def tearDownModule():
+    base.stopServer()
+
+
+class DownloadStatisticsTestCase(base.TestCase):
+
+    def setUp(self):
+        base.TestCase.setUp(self)
+
+        # Create admin user
+        admin = {'email': 'admin@email.com',
+                 'login': 'adminLogin',
+                 'firstName': 'adminFirst',
+                 'lastName': 'adminLast',
+                 'password': 'adminPassword',
+                 'admin': True}
+        self.admin = self.model('user').createUser(**admin)
+
+        self.filesDir = os.path.join(ROOT_DIR, 'plugins', 'download_statistics',
+                                     'plugin_tests', 'files')
+
+    def _downloadItem(self, itemId):
+        # Download item through REST api
+        path = '/item/%s/download' % str(itemId)
+        resp = self.request(path, user=self.admin, isJson=False)
+        self.assertStatusOk(resp)
+
+        # Iterate through generator to trigger download events
+        for data in resp.body:
+            data
+
+    def _downloadFile(self, fileId):
+        # Download file through REST api
+        path = '/file/%s/download' % str(fileId)
+        resp = self.request(path, user=self.admin, isJson=False)
+        self.assertStatusOk(resp)
+
+        # Iterate through generator to trigger download events
+        for data in resp.body:
+            data
+
+    def _checkDownloadsCount(self, fileId, count):
+        # Downloads file info and asserts download statistics are accurate
+        path = '/file/' + str(fileId)
+        resp = self.request(path, isJson=True)
+        self.assertStatusOk(resp)
+        data = resp.json
+
+        # The generator is never iterated as to not trigger additional events
```

Comment:
After making the download request, it appears as though another file download has happened, which would mean the download stats would all be one higher. Since this isn't iterating over the returned generator, another download does not happen.